File: JavaScript/Electron/todo/src/js/MenuModel.js
const tabMenuData = require('./tabMenuData')

class TabMenu {
    constructor(data) {
        this.text = data.text ? data.text : "";
        this.isActive = data.isActive != null ? data.isActive : false;
    }
}

class MenuModel {
    constructor(tabsData = tabMenuData) {
        this.tabList = tabsData.map(tabData => new TabMenu(tabData))
        this.activeTab = this.tabList.find(tab => tab.isActive) || this.tabList[0]
    }

    // findActiveTab() {
    //     return this.tabList.find(tab => tab.isActive) || this.tabList[0]
    // }

    setActiveTab(tabItem) {
        this.tabList.forEach(tab => {
            tab.isActive = (tab.text === tabItem.text);
        });
        this.activeTab = tabItem;
    }

    getActiveTab() {
        return this.activeTab;
    }

    getTabList() {
        return this.tabList;
    }
}

module.exports = {
    MenuModel
};

File: C#/GTK#/TodoApp/src/Models/MenuModel.cs
namespace TodoApp.Models;

public class MenuModel
{
    private readonly List<TabMenu> _tabList = new()
    {
        new TabMenu { Text = "All",       Filter = TabFilter.All,       IsActive = true  },
        new TabMenu { Text = "Active",    Filter = TabFilter.Active,    IsActive = false },
        new TabMenu { Text = "Completed", Filter = TabFilter.Completed, IsActive = false },
    };

    public IReadOnlyList<TabMenu> GetTabList() => _tabList.AsReadOnly();

    public TabMenu GetActiveTab() => _tabList.First(t => t.IsActive);

    public void SetActiveTab(TabMenu item)
    {
        foreach (var t in _tabList) t.IsActive = false;
        item.IsActive = true;
    }
}